app/code/Magento/EncryptionKey/Setup/Patch/Data/SodiumChachaPatch.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Magento\EncryptionKey\Setup\Patch\Data;
+
+use Magento\Framework\Setup\Patch\DataPatchInterface;
+
+/**
+ * Migrate encrypted configuration values to the latest cipher
+ */
+class SodiumChachaPatch implements DataPatchInterface
+{
+    /**
+     * @var \Magento\Framework\Setup\ModuleDataSetupInterface
+     */
+    private $moduleDataSetup;
+
+    /**
+     * @var \Magento\Config\Model\Config\Structure
+     */
+    private $structure;
+
+    /**
+     * @var \Magento\Framework\Encryption\EncryptorInterface
+     */
+    private $encryptor;
+
+    /**
+     * @var \Magento\Framework\Config\ScopeInterface
+     */
+    private $scope;
+
+    /**
+     * @param \Magento\Framework\Setup\ModuleDataSetupInterface $moduleDataSetup
+     * @param \Magento\Config\Model\Config\Structure\Proxy $structure
+     * @param \Magento\Framework\Encryption\EncryptorInterface $encryptor
+     * @param \Magento\Framework\Config\ScopeInterface $scope
+     */
+    public function __construct(
+        \Magento\Framework\Setup\ModuleDataSetupInterface $moduleDataSetup,
+        \Magento\Config\Model\Config\Structure\Proxy $structure,
+        \Magento\Framework\Encryption\EncryptorInterface $encryptor,
+        \Magento\Framework\Config\ScopeInterface $scope
+    ) {
+        $this->moduleDataSetup = $moduleDataSetup;
+        $this->structure = $structure;
+        $this->encryptor = $encryptor;
+        $this->scope = $scope;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function apply()
+    {
+        $this->moduleDataSetup->startSetup();
+
+        $this->reEncryptSystemConfigurationValues();
+
+        $this->moduleDataSetup->endSetup();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getDependencies()
+    {
+        return [];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAliases()
+    {
+        return [];
+    }
+
+    private function reEncryptSystemConfigurationValues()
+    {
+        $currentScope = $this->scope->getCurrentScope();
+
+        $this->scope->setCurrentScope(\Magento\Framework\App\Area::AREA_ADMINHTML);
+
+        try {
+            $paths = $this->structure->getFieldPathsByAttribute(
+                'backend_model',
+                \Magento\Config\Model\Config\Backend\Encrypted::class
+            );
+        } catch (\Exception $e) {
+            // This is thrown during initial application installation
```

Comment:
It's not allowed to use exceptions for implementing business logic. I propose to fix https://github.com/magento-engcom/php-7.2-support/blob/9f76f7cce2773c6367bda42a0d54dd50efbcee52/app/code/Magento/Config/Model/Config/Structure.php#L284 , so it doesn't break when the index is absent.